File: gcd.pl
#!/usr/bin/env perl

# Usage:
# $ perl gcd.pl 11 22 33 121
# 11

use strict;
use warnings;
use utf8;
use integer;
use List::Util qw/ reduce /;

sub gcd2 {
  my ( $a, $b ) = @_;
  $b == 0 ? $a : gcd2( $b, $a % $b );
}

# http://stackoverflow.com/questions/1490505/how-do-i-prevent-listmoreutils-from-warning-about-using-a-and-b-only-once
sub gcdn {
  our ( $a, $b );
  reduce { gcd2( $a, $b ) } 0, @_;
}

print gcdn(@ARGV), "\n";

File: gcd.r
#!/usr/bin/env Rscript

# R - https://www.r-project.org/
#
# Usage:
# ./gcd.r 11 22 33 121
# [1] 11
#
# Or:
# Rscript ./gcd.r 11 22 33 121
# [1] 11
#

gcd <- function (a, b) {
  if (b == 0) {
    a
  } else {
    gcd(b, a %% b)
  }
}

gcdn <- function(ns) {
  Reduce(gcd, ns)
}

args <- commandArgs(trailingOnly=TRUE)
ns <- mapply(as.integer, args)
gcdn(ns)
